### PR TITLE
Limited Table View search

### PR DESCRIPTION
**Pull request by chauhanparth210**
solve : #423 

### What shoud I do ?
I take a reference of radio button and on the basis of that value I optimize the global search (upper right hand side) .

### Screenshot

- With global search :  you can able to search all that columns 
Screenshot from 2020-03-24 00-02-26
 In above example "Language"  field is not avalible but  search is giving a result.

- With selective search : you can search only on that column that are visible 
 "Language" field is avalible
 Screenshot from 2020-03-24 00-01-58

 - "Language" field is not avalible
   Screenshot from 2020-03-24 00-02-16
 In that above case search is not showing any row

### CODE REVIEW

File: assets/app/js/controllers/scanDataTable.js [hunk 1]
```diff
@@ -340,18 +343,14 @@ class ScanDataTable extends Controller {
...
               ]
             };
-          } else if (exactFilterColumnNames.includes(columnName)) {
-            query.where.$and[columnName] = {
-              $eq: columnSearch
-            };
           } else {
             query.where.$and[columnName] = {
-              $like: `%${columnSearch}%`
+              $eq: columnSearch
```
Comment (steven-esser) [inline assets/app/js/controllers/scanDataTable.js:348]: Seem my comments on your other PR

File: assets/app/js/controllers/scanDataTable.js [hunk 2]
```diff
@@ -360,10 +359,20 @@ class ScanDataTable extends Controller {
...
+                $like: Sequelize.literal(`"%${escapedSearch}%" ESCAPE '\\'`)
+              };
+              query.where.$and.$or.push(orSearch);  
+            }
+          } else {
+            orSearch[dataTablesInput.columns[i].name] = {
+              $like: Sequelize.literal(`"%${escapedSearch}%" ESCAPE '\\'`)
+            };
+            query.where.$and.$or.push(orSearch);  
+          }
```
Comment (steven-esser) [inline assets/app/js/controllers/scanDataTable.js:375]: Have you tested any of this?

**Review by steven-esser — CHANGES_REQUESTED** [review]

File: assets/app/js/controllers/scanDataTable.js (hunk 2 above)
Comment (chauhanparth210) [reply assets/app/js/controllers/scanDataTable.js:375]: No currently I am not tested any of this. But I checked manually it works fine as you see on the screen shorts.

**Review by nitin10s — COMMENTED** [review]
Selective Search and Global Search could be unclear, I'll suggest a checkbox with Label: "Search across all columns".

If unchecked state: Selective Search would take place.
If checked state: Global Search

**Comment by steven-esser** [comment]
@chauhanparth210 Ping. I will close this soon if you have no more updates.

**Comment by steven-esser** [comment]
Closing due to stagnation.

**Pull request closed without merging**